mediator/src/controllers/service-request.ts

```diff
@@ -1,95 +1,43 @@
-import axios from 'axios';
-import { logger } from '../../logger';
-import { FHIR, CHT } from '../../config';
-import { generateFHIRSubscriptionResource } from '../utils/fhir';
-import https from 'https';
-import { generateChtRecordsApiUrl } from '../utils/url';
-
-const { url: fhirUrl, password: fhirPassword, username: fhirUsername } = FHIR;
-
-interface IServiceRequest {
-  patient_id: string;
-  callback_url: string;
-}
-
-export async function createServiceRequest(request: IServiceRequest) {
+import { logger } from "../../logger";
+import { createChtRecord } from "../utils/cht";
+import {
+  getFHIRPatientResource,
+  getFHIROrgEndpointResource,
+  createFHIRSubscriptionResource,
+  deleteFhirSubscription,
+} from "../utils/fhir";
+
+export async function createServiceRequest(request: fhir4.ServiceRequest) {
   try {
-    const {patient_id: patientId, callback_url: callbackUrl} = request;
-
-    const patientRes = await getFHIRPatientResource(patientId); 
+    const patientId = (request.subject as any).reference.replace("Patient/", "");
+    // checking if patient exists - axios throws error for non 200
+    await getFHIRPatientResource(patientId);
 
-    if (patientRes.status !== 200) {
-      return { status: patientRes.status, data: patientRes.data };
-    }
+    const orgId = (request.requester as any).reference.replace("Organization/", "");
+    const endpointRes = await getFHIROrgEndpointResource(orgId);
 
-    // generate subscription resource
-    const subRes = await createFHIRSubscriptionResource(patientId, callbackUrl);
-
-    if (subRes.status !== 201) {
-      return { status: subRes.status, data: subRes.data };
-    }
-
-    // call the CHT API to set up the follow up task
+    // Generate subscription resource
```

Comment:
Is this comment necessary, as it basically explains the what, which is already clear in the `createFHIRSubscriptionResource` function's name. Please replace it with **why**, or remove it.